torch/_inductor/compile_fx_async.py

```diff
@@ -179,3 +179,183 @@ def callback(pickled_output: _WireProtocolPickledOutput) -> OutputCode:
             return output.graph
 
         return _AsyncOutputCode(eager_output_code, f, callback)
+
+
+# _ProgressiveOutputCode handles running a fast compile first, then hot-swapping
+# to a more optimized version when the expensive compile finishes.
+@final
+class _ProgressiveOutputCode(OutputCode):
+    _fast_output_code: Optional[OutputCode]
+    _optimized_output_code: Optional[OutputCode]
+    _progression_futures: list[Future[_WireProtocolPickledOutput]]
+    _callback: Callable[[_WireProtocolPickledOutput], OutputCode]
+    _post_compile_data: Optional[_PostCompileData] = None
+    _boxed_call: bool
+    _current_progression_index: int
+
+    def __init__(
+        self,
+        # Fast compile that runs faster than the progressive compiles
+        fast_output_code: OutputCode,
+        # Futures for the progressive optimized compiles
+        progression_futures: list[Future[_WireProtocolPickledOutput]],
+        # Callback to convert the optimized result to OutputCode
+        callback: Callable[[_WireProtocolPickledOutput], OutputCode],
+    ) -> None:
+        self._fast_output_code = fast_output_code
+        self._optimized_output_code = None
+        self._progression_futures = progression_futures
+        self._callback = callback
+        self._boxed_call = getattr(fast_output_code, "_boxed_call", False)
+        self._current_progression_index = -1
+
+    @override
+    def __call__(self, *args: Any) -> Any:
+        # Check if any newer progression stage is ready and switch to it
+        args = self._check_and_switch_progression(args)
+
+        if self._optimized_output_code is not None:
```

Comment:
Hm. It seems like the only reason for the split between `_optimized_output_code` and `_fast_output_code` is to track stats. Maybe instead have a single `_output_code` and a separate `_stat_name`